src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/TopLevelCommandManager.cs

```diff
@@ -49,6 +60,22 @@ private async Task LoadTopLevelCommandsFromProvider(CommandProviderWrapper comma
         }
     }
 
+    public async Task ReloadAllCommandsAsync()
+    {
+        // Oh dear this may have reavealed the awful truth that we load
+        // extensions on the UI thread.
+        //
+        // If we stick this in a BG task, then it explodes, because
+        // TopLevelCommands is Observable, so adding things to it can only
```

Comment:
Should we just give the TLCM here the TaskScheduler and be dispatching the add to TopLevelCommands only to the UI thread so we can do everything else here in the background?

Reply:
I'll try out a commit and see what it feels like to have the commands loaded in the background. Predictably, I think that's gonna be about its own PR worth of work

Reply:
oh my god I'm so glad I did. BG extension loading is so much better